staging/src/k8s.io/apiserver/pkg/storage/testing/storage.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apiserver/pkg/apis/example"
+	"k8s.io/apiserver/pkg/storage"
+)
+
+type ObjectInvariantsFunc func(context.Context, *testing.T, storage.Interface, string)
+
+func StorageInterfaceCreateTest(t *testing.T, store storage.Interface, checkInvariants ObjectInvariantsFunc) {
+	ctx := context.Background()
+	key := "/testkey"
+	out := &example.Pod{}
+	obj := &example.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo", SelfLink: "testlink"}}
+
+	// verify that object doesn't exist before set.
+	if err := store.Get(ctx, key, storage.GetOptions{}, out); !storage.IsNotFound(err) {
+		t.Errorf("no object expected, got: %#v, err: %v", out, err)
+	}
+
+	if err := store.Create(ctx, key, obj, out, 0); err != nil {
+		t.Fatalf("Create failed: %v", err)
+	}
+	// basic tests of the output
+	if obj.ObjectMeta.Name != out.ObjectMeta.Name {
+		t.Errorf("pod name want=%s, get=%s", obj.ObjectMeta.Name, out.ObjectMeta.Name)
+	}
+	if out.ResourceVersion == "" {
+		t.Errorf("output should have non-empty resource version")
+	}
+	if out.SelfLink != "" {
+		t.Errorf("output should have empty selfLink")
+	}
+	checkInvariants(ctx, t, store, key)
```

Comment:
in addition to whatever `checkInvariants` does, I'd have expected `StorageInterfaceCreateTest` to directly check the object now exists via Get and produces an identically populated `obj`

Reply:
Get() is happening above (line 39) - this was an additional check (that etcd3 implementation can use to access etcd directly using etcd API).

But maybe that's an overkill if we have Get above anyway.